File: frontend/src/dm/obsolete/schemas.ts
/** 已廢止文件查詢（US10 / UCDM08 / DM06）型別（對齊後端 app/dm/obsolete_archive/schemas.py）。 */

/** 已廢止文件清單列（含末版版號 + 廢止脈絡）。原作者採末版〔在架版〕作者（SA 裁示 Q2=B）。 */
export interface ObsoleteDocItem {
  doc_id: string
  doc_name: string
  latest_version_no: string | null
  category_code: string
  category_name: string
  author_id: string | null
  author_name: string | null
  obsolete_date: string | null
  applicant_id: string | null
  applicant_name: string | null
  approver_id: string | null
  approver_name: string | null
  obsolete_reason: string | null
}

/** DM06 入口可見性（供側欄逐項閘）。 */
export interface ObsoleteAccess {
  can_access: boolean
}

/** 查詢條件（關鍵字＝文件名 / 廢止原因；分類；廢止日期區間）。 */
export interface ObsoleteFilters {
  keyword: string
  category: string // '' = 全部
  dateFrom: string
  dateTo: string
}

export const EMPTY_OBSOLETE_FILTERS: ObsoleteFilters = {
  keyword: "",
  category: "",
  dateFrom: "",
  dateTo: "",
}

File: backend/app/dm/obsolete_archive/schemas.py
"""已廢止文件查詢 schema（US10 / UCDM08 / DM06）。"""

from datetime import date, datetime

from pydantic import BaseModel


class ObsoleteQuery(BaseModel):
    """查詢條件：關鍵字（文件名 / 廢止原因）、分類、廢止日期區間（比對核准廢止之完成時間）。"""

    keyword: str | None = None
    category: str | None = None
    date_from: date | None = None
    date_to: date | None = None


class ObsoleteDocItem(BaseModel):
    """已廢止文件清單列（FR-003）。原作者採末版〔在架版〕作者（SA 裁示 Q2=B）。"""

    doc_id: str
    doc_name: str
    latest_version_no: str | None  # 末版版號
    category_code: str
    category_name: str
    author_id: str | None  # 末版作者（DM_DOC_VERSION.CREATED_USER）
    author_name: str | None
    obsolete_date: datetime | None  # 廢止時間＝核准廢止之完成時間
    applicant_id: str | None  # 廢止申請人＝OBSOLETE 週期之 CREATED_USER
    applicant_name: str | None
    approver_id: str | None  # 核准者
    approver_name: str | None
    obsolete_reason: str | None


class ObsoleteAccess(BaseModel):
    """DM06 入口可見性（FR-001，供前端側欄逐項閘；後端另有硬閘擋直連）。"""

    can_access: bool
